### PR TITLE
bpf: xdp asm volatile fix in relation to reg spill

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg.

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Review by jrfastab — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by borkmann** [comment]
test-focus K8sService.*

**Comment by borkmann** [comment]
restart-ginkgo

**Comment by borkmann** [comment]
restart-gke

**Review by pchaigno — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Comment by qmonnet** [comment]
test-focus K8sService.*

**Comment by qmonnet** [comment]
restart-gke

**Comment by qmonnet** [comment]
test-gke

File: daemon/cmd/daemon_main.go [hunk 1]
```diff
@@ -1580,6 +1580,10 @@ func initKubeProxyReplacementOptions() {
 
 	if option.Config.EnableNodePort &&
 		option.Config.NodePortAcceleration != option.NodePortAccelerationNone {
+		if option.Config.Tunnel != option.TunnelDisabled {
+			log.Fatal("Cannot use NodePort acceleration with tunneling. Only direct routing mode supported at this point.")
```
Comment (aanm) [inline daemon/cmd/daemon_main.go]: To make it clear for users how they can fix this we should probably end this error message with `supported at this point. Either run cilium-agent with '--%s=%s' or '--%s=%s'", option.NodePortRange, option.NodePortAccelerationNone, option.TunnelName, option.TunnelDisabled)`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**